primeorder/src/projective.rs

```diff
@@ -258,13 +258,24 @@ where
 
 impl<C> Group for ProjectivePoint<C>
 where
-    Self: Double,
     C: PrimeCurveParams,
+    FieldBytes<C>: Copy,
 {
     type Scalar = Scalar<C>;
 
     fn try_from_rng<R: TryRngCore + ?Sized>(rng: &mut R) -> core::result::Result<Self, R::Error> {
-        Ok(Self::GENERATOR * <Scalar<C> as Field>::try_from_rng(rng)?)
+        let mut bytes = FieldBytes::<C>::default();
+        let mut sign = 0;
+
+        loop {
+            rng.try_fill_bytes(&mut bytes)?;
+            rng.try_fill_bytes(core::array::from_mut(&mut sign))?;
+            if let Some(point) =
+                AffinePoint::decompress(&bytes, Choice::from(sign & 1)).into_option()
+            {
+                return Ok(point.into());
+            }
+        }
```

Comment:
I know I just stamped approve, but now I'm kind of noticing that perhaps this should be an inherent `pub fn try_from_rng` impl'd on `AffinePoint`, and then the `ProjectivePoint` impl could be:

```suggestion
        AffinePoint::try_from_rng(rng).map(Into::into)
```

Reply:
I agree, but don't we need a trait for that to implement it in `primeorder`?

Reply:
Apparently not.
Working on it.

Reply:
Done.

`EdwardsPoint` and `DecafPoint` still need a bit more work.
After this is merged I will do the necessary follow-up for `EdwardsPoint` in #1333.
Will make a PR for `DecafPoint` later.